Objects/dictobject.c

```diff
@@ -3651,6 +3668,210 @@ PyTypeObject PyDictIterItem_Type = {
 };
 
 
+/* dictreviter */
+
+static PyObject *
+dictreviter_iternextkey(dictiterobject *di)
+{
+    PyObject *key;
+    Py_ssize_t i;
+    PyDictKeysObject *k;
+    PyDictObject *d = di->di_dict;
```

Comment:
Since we use C11 from Python 3.6, I prefer [DeclareVariablesAtFirstUse](http://wiki.c2.com/?DeclareVariablesAtFirstUse) style for new code.